src/Microsoft.OData.Core/ODataBatchOperationRequestMessage.cs

```diff
@@ -66,9 +84,13 @@ public sealed class ODataBatchOperationRequestMessage : IODataRequestMessage, IO
             this.Method = method;
             this.Url = requestUrl;
             this.ContentId = contentId;
+            this.groupId = groupId;
 
             this.message = new ODataBatchOperationMessage(contentStreamCreatorFunc, headers, operationListener, payloadUriConverter, writing);
             this.Container = container;
+            this.dependsOnIds = dependsOnIds != null
+                             ? new List<string>(dependsOnIds)
+                             : null;

```

Comment:
If the caller passes in an empty list, we will set dependsOnIds to an empty list.  If the caller passes in null we will create a null dependsOnIds.  That means that the public caller will have to check for both conditions. Why not just always create a List<string> containing any values passed in to the constructor?

Reply:
updated to use List<string> and exposed as read-only getter.

---
In reply to: [155349744](https://github.com/OData/odata.net/pull/992#discussion_r155349744) [](ancestors = 155349744)

Reply:
We should still only require that the caller pass in an IEnumerable<string>, rather than an IList<string>.  We copy it to our list (as we should) so it doesn't have to be a list. The rest is of the code is fine.

---
In reply to: [155390374](https://github.com/OData/odata.net/pull/992#discussion_r155390374) [](ancestors = 155390374,155349744)

Reply:
Restored to IEnumerable for the argument type. Thanks.

---
In reply to: [156760101](https://github.com/OData/odata.net/pull/992#discussion_r156760101) [](ancestors = 156760101,155390374,155349744)